app/models/ext_management_system.rb

```diff
@@ -33,20 +33,16 @@ def self.permitted?
   end
   delegate :permitted?, :to => :class
 
-  def self.supported_types_and_descriptions_hash
-    supported_subclasses.each_with_object({}) { |klass, hash| hash[klass.ems_type] = klass.description }
-  end
-
   def self.subclasses_supports?(feature)
-    supported_subclasses.select { |subclass| subclass.supports?(feature) }
+    permitted_subclasses.select { |subclass| subclass.supports?(feature) }
   end
 
   def self.api_allowed_attributes
     %w[]
   end
 
   def self.supported_types_for_create
```

Comment:
Note that I kept "supported" as the term for things where it is ultimately applies a supports_feature check.  This one is a great example of that distiction.